hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -297,15 +298,15 @@ public void handleQuery(@NonNull final Bytes requestBuffer, @NonNull final Buffe
     private Query parseQuery(Bytes requestBuffer) {
         try {
             return queryParser.parseStrict(requestBuffer.toReadableSequentialData());
-        } catch (MalformedProtobufException | UnknownFieldException e) {
-            throw new StatusRuntimeException(Status.INVALID_ARGUMENT);
-        } catch (IOException e) {
-            // This should technically not be possible. The data buffer supplied
-            // is either based on a byte[] or a byte buffer, in both cases all data
-            // is available and a generic IO exception shouldn't happen. If it does,
-            // it indicates the data could not be parsed, but for a reason other than
-            // those causing an MalformedProtobufException or UnknownFieldException.
-            logger.warn("Unexpected IO exception while parsing protobuf", e);
+        } catch (ParseException e) {
+            switch (e.getCause()) {
+                case MalformedProtobufException ex:
+                    break;
```

Comment:
We should probably throw exception in both these cases here to keep same behavior
```
throw new StatusRuntimeException(Status.INVALID_ARGUMENT);
```

Reply:
This exception is thrown in all of these cases right below the `switch()` at line 310. This `switch` statement is only used to decide if we want to log a warning. It's coded to implement the exact same logic as was present before with individual `catch()` clauses.

Reply:
ah yes ! makes sense. Thanks @anthony-swirldslabs